url6a9f553762_Bresiu_socket_io_tgz-pJ8-MainJ8/src/com/github/nkzawa/hasbinarydata/HasBinaryData.java

```diff
@@ -0,0 +1,57 @@
+package com.github.nkzawa.hasbinarydata;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+public class HasBinaryData {
+
+    private HasBinaryData() {}
+
+    public static boolean hasBinary(Object data) {
+        return recursiveCheckForBinary(data);
+    }
+
+    private static boolean recursiveCheckForBinary(Object obj) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.